agent/app/agent_unix.go

```diff
@@ -0,0 +1,156 @@
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package app
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/aws/amazon-ecs-agent/agent/ec2"
+	"github.com/aws/amazon-ecs-agent/agent/ecscni"
+	"github.com/aws/amazon-ecs-agent/agent/engine"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
+	"github.com/aws/amazon-ecs-agent/agent/eni/pause"
+	enisetup "github.com/aws/amazon-ecs-agent/agent/eni/setup"
+)
+
+// initPID defines the process identifier for the init process
+const initPID = 1
+
+// awsVPCCNIPlugins is a list of CNI plugins required by the ECS Agent
+// to configure the ENI for a task
+var awsVPCCNIPlugins = []string{ecscni.ECSENIPluginName,
+	ecscni.ECSBridgePluginName,
+	ecscni.ECSIPAMPluginName,
+}
+
+// initializeTaskENIDependencies initializes all of the dependencies required by
+// the Agent to support the 'awsvpc' networking mode. A non nil error is returned
+// if an error is encountered during this process. An additional boolean flag to
+// indicate if this error is considered terminal is also returned
+func (agent *ecsAgent) initializeTaskENIDependencies(state dockerstate.TaskEngineState, taskEngine engine.TaskEngine) (error, bool) {
+	// Check if the Agent process's pid  == 1, which means it's running without an init system
+	if agent.os.Getpid() == initPID {
+		// This is a terminal error. Bad things happen with invoking the
+		// the ENI plugin when there's no init process in the pid namesapce.
+		// Specifically, the DHClient processes that are started as children
+		// of the Agent will not be reaped leading to the ENI device
+		// disappearing until the Agent is killed.
+		return errors.New("agent is not started with an init system"), true
+	}
+
+	// Set VPC and Subnet IDs for the instance
+	if err, ok := agent.setVPCSubnet(); err != nil {
+		return err, ok
+	}
+
+	// Validate that the CNI plugins exist in the expected path and that
+	// they possess the right capabilities
+	if err := agent.queryCNIPluginsCapabilities(); err != nil {
+		// An error here is terminal as it means that the plugins
+		// do not support the ENI capability
+		return err, true
+	}
+
+	// Load the pause container's image from the 'disk'
+	if _, err := agent.pauseLoader.LoadImage(agent.cfg, agent.dockerClient); err != nil {
+		if pause.IsNoSuchFileError(err) || pause.UnsupportedPlatform(err) {
+			// If the pause container's image tarball doesn't exist or if the
+			// invocation is done for an unsupported platform, we cannot recover.
+			// Return the error as terminal for these cases
+			return fmt.Errorf("unable to load pause container image: %v", err), true
+		}
+		return fmt.Errorf("unable to load pause container image: %v", err), false
```

Comment:
I did not want to use that here because I don't think an error stack trace here provides much value beyond the `fmt.Errorf` wrapping.